File: YiYuanSVN/app/src/main/java/com/iask/yiyuanlegou1/account/RegisterActivity.java
package com.iask.yiyuanlegou1.account;

import android.content.Intent;
import android.text.TextUtils;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.iask.yiyuanlegou1.R;
import com.iask.yiyuanlegou1.base.AppConfigure;
import com.iask.yiyuanlegou1.base.BaseActivity;
import com.iask.yiyuanlegou1.base.BaseApplication;
import com.iask.yiyuanlegou1.common.cache.CacheKeys;
import com.iask.yiyuanlegou1.common.cache.CacheManager;
import com.iask.yiyuanlegou1.home.HomeActivity;
import com.iask.yiyuanlegou1.log.AndroidLogger;
import com.iask.yiyuanlegou1.log.Logger;
import com.iask.yiyuanlegou1.network.IOpenApiDataServiceCallback;
import com.iask.yiyuanlegou1.network.respose.CommonResponse;
import com.iask.yiyuanlegou1.network.respose.account.RegisterResponse;
import com.iask.yiyuanlegou1.network.service.account.RegisterService;
import com.iask.yiyuanlegou1.network.service.account.VerifyCodeService;
import com.iask.yiyuanlegou1.utils.StringUtils;
import com.iask.yiyuanlegou1.widget.TitleBarView;


public class RegisterActivity extends BaseActivity implements OnClickListener {
    private TitleBarView title;
    private EditText registerU, registerP, vericodeE;
    private Button btngetIdcode, btnSubmit;
    private String mobile;
    private Logger logger = AndroidLogger.getLogger(RegisterActivity.class);

    @Override
    protected void findViewById() {
        title = (TitleBarView) findViewById(R.id.title);
        registerU = (EditText) findViewById(R.id.register_u);
        registerP = (EditText) findViewById(R.id.register_p);
        vericodeE = (EditText) findViewById(R.id.vericode);

        btngetIdcode = (Button) findViewById(R.id.getIdcode);
        btnSubmit = (Button) findViewById(R.id.register_btn);

        btngetIdcode.setOnClickListener(this);
        btnSubmit.setOnClickListener(this);
    }

    @Override
    protected void initView() {
        title.setCommonTitle(View.VISIBLE, View.VISIBLE);
        title.setBtnLeft(R.mipmap.back, R.string.whitespace);
        title.setBtnLeftOnclickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                RegisterActivity.this.finish();
            }
        });
        title.setTitleText(R.string.register);
    }

    @Override
    protected int getContentLayout() {
        return R.layout.activity_register;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.register_btn:
                doRegister();
                break;
            case R.id.getIdcode:
                doGetVeriCode();
                break;
        }
    }

    private void doRegister() {
        String pwd = registerP.getText().toString();
        if (TextUtils.isEmpty(pwd)) {
            Toast.makeText(RegisterActivity.this, "密码不能为空！", Toast.LENGTH_SHORT).show();
            return;
        }
        String veriCode = vericodeE.getText().toString();
        if (TextUtils.isEmpty(veriCode)) {
            Toast.makeText(RegisterActivity.this, "验证码不能为空！", Toast.LENGTH_SHORT).show();
            return;
        }
        RegisterService service = new RegisterService(this);
        service.setCallback(new IOpenApiDataServiceCallback<RegisterResponse>() {
            @Override
            public void onGetData(RegisterResponse data) {
                try {
                    CacheManager.saveObject(CacheManager.TYPE_USER_INFO, CacheKeys
                            .USERINFO_LOGINVO, data.data.user);
                    CacheManager.saveObject(CacheManager.TYPE_USER_INFO, CacheKeys
                            .OSS_SERVER_INFO, data.data.aliyun);
                    CacheManager.saveString(CacheManager.TYPE_PUBLIC, CacheKeys
                            .APPDATA_CURR_REGIONURL, data.data.aliyun.getRegionUrl());
                    BaseApplication.getInstance().setUserId(data.data.user.getUserId() + "");
                    Toast.makeText(RegisterActivity.this, "注册成功！", Toast.LENGTH_SHORT).show();
                    startActivity(new Intent(RegisterActivity.this, HomeActivity.class));
                    RegisterActivity.this.finish();
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }

            @Override
            public void onGetError(int errorCode, String errorMsg, Throwable error) {
                Toast.makeText(RegisterActivity.this, errorMsg, Toast.LENGTH_SHORT).show();
            }
        });
        service.post("phone=" + mobile + "&pwd=" + pwd + "&code=" + veriCode, true);
    }

    private void doGetVeriCode() {
        mobile = registerU.getText().toString();
        if (TextUtils.isEmpty(mobile)) {
            Toast.makeText(RegisterActivity.this, "手机号码不能为空！", Toast.LENGTH_SHORT).show();
            return;
        }
        if (!StringUtils.isMobileNO(mobile)) {
            Toast.makeText(RegisterActivity.this, "手机号码格式不正确！", Toast.LENGTH_SHORT).show();
            return;
        }
        VerifyCodeService service = new VerifyCodeService(this);
        service.setCallback(new IOpenApiDataServiceCallback<CommonResponse>() {
            @Override
            public void onGetData(CommonResponse data) {
                logger.debug(data.getMsg());
                Toast.makeText(RegisterActivity.this, data.getMsg(), Toast.LENGTH_SHORT).show();
                if (!AppConfigure.officialVersion) {
                    vericodeE.setText(data.getMsg());
                }
            }

            @Override
            public void onGetError(int errorCode, String errorMsg, Throwable error) {
                logger.debug(errorMsg);
                Toast.makeText(RegisterActivity.this, errorMsg, Toast.LENGTH_SHORT).show();
            }
        });
        service.post("phone=" + mobile + "&type=0", true);
    }
}

File: Eyepetizer-in-Kotlin-master/app/src/main/java/com/tt/lvruheng/eyepetizer/ui/RegisterActivity.kt
package com.tt.lvruheng.eyepetizer.ui

import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.InputType
import android.text.TextUtils
import android.text.TextWatcher
import android.view.View
import android.widget.EditText
import com.tt.lvruheng.eyepetizer.R
import com.tt.lvruheng.eyepetizer.common.cache.CacheKeys
import com.tt.lvruheng.eyepetizer.common.cache.CacheManager
import com.tt.lvruheng.eyepetizer.network.IOpenApiDataServiceCallback
import com.tt.lvruheng.eyepetizer.network.ServiceURL
import com.tt.lvruheng.eyepetizer.network.response.RegisterResponse
import com.tt.lvruheng.eyepetizer.network.service.RegisterService
import com.tt.lvruheng.eyepetizer.ui.dialog.VerifyFragment
import com.tt.lvruheng.eyepetizer.utils.showToast
import kotlinx.android.synthetic.main.activity_register.*
import kotlinx.android.synthetic.main.title_style_1.*

class RegisterActivity : BaseActivity() {

    override fun getLayoutId(): Int {
        return R.layout.activity_register
    }

    override fun initView() {
        var mobileNum = intent.getStringExtra("mobileNum")
        tvTitle.text = "注册"
        ivBack.setOnClickListener({
            finish()
        })
        btnNext.setOnClickListener({
            val pwd = etPwd.text.toString()
            val code = etOldPwd.text.toString()
            val registerService = RegisterService(this)
            registerService.callback = object : IOpenApiDataServiceCallback<RegisterResponse> {

                override fun onGetData(data: RegisterResponse?) {
                    if (data!!.data.login.equals("0")) {
                        showToast("注册成功")
                        CacheManager.saveObject(CacheManager.TYPE_USER_INFO, CacheKeys.USERINFO_LOGINVO, data!!.data)
                        var intent = Intent()
                        intent.setClass(this@RegisterActivity, MainActivity::class.java)
                        startActivity(intent)
                    } else {
                        showToast(data.retMsg)
                    }
                }

                override fun onGetError(errorCode: Int, errorMsg: String?, error: Throwable?) {
                    showToast(errorMsg!!)
                }

            }
            registerService.postLogined("mobileNo=${mobileNum}&passwd=${pwd}&verifyCode=${code}", true)
        })
        btnGetCode.setOnClickListener({
            val dialog = VerifyFragment()
            dialog.show(supportFragmentManager, "veryCode")
            val data = Bundle()
            data.putString("mobileNum", mobileNum)
            dialog.arguments = data
            dialog.setOnBackListener(object : VerifyFragment.OnBackListener {
                override fun onBack(code: Int) {
                    if (code == 0) {
                        btnGetCode.startTimer(60 * 1000)
                    }
                }
            })
        })
        etPwd.setTextChangeListener {
            btnNext.isEnabled = !TextUtils.isEmpty(etOldPwd.text) && !TextUtils.isEmpty(it) && it.length >= 6
            if (!TextUtils.isEmpty(it)) {
                ivClear.visibility = View.VISIBLE
                ivEye.visibility = View.VISIBLE
            } else {
                ivClear.visibility = View.GONE
            }
        }
        ivEye.setOnClickListener({
            if (etPwd.inputType != InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD) {
                etPwd.inputType = (InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD)
                ivEye.setImageResource(R.drawable.eye_open)
            } else {
                etPwd.inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
                ivEye.setImageResource(R.drawable.eye_close)
            }
        })
        ivClear.setOnClickListener({
            etPwd.setText("")
        })
        tvProtocal.setOnClickListener({
            val intent = Intent()
            intent.setClass(this@RegisterActivity, WebViewActivity::class.java)
            intent.putExtra("url", "${ServiceURL.URL}#/protocol/register")
            intent.putExtra("showTitle", true)
            intent.putExtra("title", "注册协议")
            startActivity(intent)
        })
    }

    private fun EditText.setTextChangeListener(body: (key: String) -> Unit) {
        addTextChangedListener(object : TextWatcher {
            override fun afterTextChanged(s: Editable?) {
            }

            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
            }

            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                body(s.toString())
            }
        })
    }
}
